cocos/renderer/gfx-base/GFXBuffer.h

```diff
@@ -37,11 +37,12 @@ class CC_DLL Buffer : public GFXObject {
 
     static uint computeHash(const BufferInfo &info);
 
-    virtual bool initialize(const BufferInfo &info)     = 0;
-    virtual bool initialize(const BufferViewInfo &info) = 0;
-    virtual void destroy()                              = 0;
-    virtual void resize(uint size)                      = 0;
-    virtual void update(void *buffer, uint size)        = 0;
+    void initialize(const BufferInfo &info);
+    void initialize(const BufferViewInfo &info);
```

Comment:
Why change return type to void?

Reply:
When running asynchronously in different threads, there is no way to get any meaningful return values. We should insert more asserts and build a much more strict and versatile independent validation layer instead.

Reply:
Sounds reasonable. Then i think we should modify all gfx classes.